librato.go

```diff
@@ -29,33 +31,48 @@ type Reporter struct {
 	Percentiles     []float64              // percentiles to report on histogram metrics
 	TimerAttributes map[string]interface{} // units in which timers will be displayed
 	intervalSec     int64
+	done            <-chan struct{}
 }
 
-func NewReporter(r metrics.Registry, d time.Duration, e string, t string, s string, p []float64, u time.Duration) *Reporter {
-	return &Reporter{e, t, s, d, r, p, translateTimerAttributes(u), int64(d / time.Second)}
+func NewReporter(ctx context.Context, r metrics.Registry, d time.Duration, e string, t string, s string, p []float64, u time.Duration) *Reporter {
+	return &Reporter{e, t, s, d, r, p, translateTimerAttributes(u), int64(d / time.Second), ctx.Done()}
 }
 
-func Librato(r metrics.Registry, d time.Duration, e string, t string, s string, p []float64, u time.Duration) {
-	NewReporter(r, d, e, t, s, p, u).Run()
+func Librato(ctx context.Context, r metrics.Registry, d time.Duration, e string, t string, s string, p []float64, u time.Duration) {
+	NewReporter(ctx, r, d, e, t, s, p, u).Run()
```

Comment:
Do we need to store anything? Just make `Run()` take a context. Then, Librato(ctx, ...), since it calls run, and NewReporter() stays the same. Can then `case <-ctx.Done():` instead.